web/packages/teleport/src/Users/UserAddEdit/useDialog.tsx

```diff
@@ -91,3 +106,22 @@ export type Props = {
   onCreate(user: User): Promise<any>;
   onUpdate(user: User): Promise<any>;
 };
+
+export function traitsToTraitsOption(allTraits: AllUserTraits): TraitsOption[] {
+  const newTrait = [];
+  for (let trait in allTraits) {
+    if (!allTraits[trait][0]) {
```

Comment:
actually, can we add a test for this function? wouldn't this throw if `allTraits[trait] = null`?
b/c i went to my staging cluster and saw this in my traits:
<img width="516" alt="image" src="https://github.com/gravitational/teleport/assets/43280172/0aa6af1d-9916-4faf-b518-9b10ed80b428">

Reply:
for this condition `!allTraits[trait][0]` can this be possible?

```js
logins: ["", "root"]
```

if valid, this trait will get ignored?

Reply:
Tests added in https://github.com/gravitational/teleport/pull/42620/commits/1d38fb380b1465b91a7b0a724628a2df3f2e63ea

- `logins: ["", "root"]` will be shown as is. But `logins: [""]` will be filtered out. I am still sticking with this choice as in the backend, in its current state, that's a still default value not set by the user. 
- that didn't throw error cause `makeTraits` checks and sets empty array. Though, just to be on safe side, https://github.com/gravitational/teleport/pull/42620/commits/1e80fcff976f4a4647a2eb6f1e1c8eabfd843d43 now avoids null value explicitly.